src/coreclr/hosts/corerun/corerun.cpp

```diff
@@ -113,51 +74,46 @@ class HostEnvironment
         // Pin the module - CoreCLR.dll does not support being unloaded.
         HMODULE dummy_coreCLRModule;
         if (!::GetModuleHandleExW(GET_MODULE_HANDLE_EX_FLAG_PIN, coreCLRPath, &dummy_coreCLRModule)) {
-            *m_log << W("Failed to pin: ") << coreCLRPath << Logger::endl;
+            *m_log << W("Failed to pin: ") << coreCLRPath.GetUnicode() << Logger::endl;
             return nullptr;
         }
 
-        wchar_t coreCLRLoadedPath[MAX_LONGPATH];
-        ::GetModuleFileNameW(result, coreCLRLoadedPath, MAX_LONGPATH);
+        StackSString coreCLRLoadedPath;
+        WszGetModuleFileName(result, coreCLRLoadedPath);
 
-        *m_log << W("Loaded: ") << coreCLRLoadedPath << Logger::endl;
+        *m_log << W("Loaded: ") << coreCLRLoadedPath.GetUnicode() << Logger::endl;
 
         return result;
     }
 
 public:
     // The path to the directory that CoreCLR is in
-    wchar_t m_coreCLRDirectoryPath[MAX_LONGPATH];
+    PathString m_coreCLRDirectoryPath;
 
     HostEnvironment(Logger *logger) 
         : m_log(logger), m_CLRRuntimeHost(nullptr) {
 
             // Discover the path to this exe's module. All other files are expected to be in the same directory.
-            DWORD thisModuleLength = ::GetModuleFileNameW(::GetModuleHandleW(nullptr), m_hostPath, MAX_LONGPATH);
+            WszGetModuleFileName(::GetModuleHandleW(nullptr), m_hostPath);
 
             // Search for the last backslash in the host path.
-            int lastBackslashIndex;
-            for (lastBackslashIndex = thisModuleLength-1; lastBackslashIndex >= 0; lastBackslashIndex--) {
-                if (m_hostPath[lastBackslashIndex] == W('\\')) {
-                    break;
-                }
-            }
+            SString::CIterator lastBackslash = m_hostPath.End();
+            m_hostPath.FindBack(lastBackslash, W('\\'));
 
             // Copy the directory path
-            ::wcsncpy_s(m_hostDirectoryPath, m_hostPath, lastBackslashIndex + 1);
+            m_hostDirectoryPath.Set(m_hostPath, m_hostPath.Begin(), lastBackslash + 1);
```

Comment:
Thanks. I didn't realize CopySystemDirectory really means "get path from file name". It seems to be mis-named.

For this particular case, I still need to find the location of the last backslash, since I need it to create m_hostExeName below. Given that, there seems to be no more advantage to use CopySystemDirectory.